### PR TITLE
Initial docker image for aws lambda feature server

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**: This PR creates an initial dockerfile for AWS lambda feature server.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/infra/feature_servers/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# Copy necessary parts of the Feast codebase
+COPY sdk/python sdk/python
+COPY protos protos
+COPY README.md README.md
+
+# Install Feast for AWS with Lambda dependencies
+RUN pip3 install -e 'sdk/python[aws,redis]'
+RUN  pip3 install -r sdk/python/feast/infra/feature_servers/aws_lambda/requirements.txt --target "${LAMBDA_TASK_ROOT}"
+
+ENV FEAST_USAGE=False
```
Comment (felixwang9817) [inline sdk/python/feast/infra/feature_servers/Dockerfile:15]: Why are we disabling usage?

File: sdk/python/feast/infra/feature_servers/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+COPY sdk/python/feast/infra/feature_servers/aws_lambda/app.py ${LAMBDA_TASK_ROOT}
+
+# Copy necessary parts of the Feast codebase
+COPY sdk/python sdk/python
+COPY protos protos
+COPY README.md README.md
+
+# Install Feast for AWS with Lambda dependencies
+RUN pip3 install -e 'sdk/python[aws,redis]'
+RUN  pip3 install -r sdk/python/feast/infra/feature_servers/aws_lambda/requirements.txt --target "${LAMBDA_TASK_ROOT}"
```
Comment (felixwang9817) [inline sdk/python/feast/infra/feature_servers/Dockerfile:13]: Nit: two spaces, should just be `RUN pip3`.

File: sdk/python/feast/infra/feature_servers/app.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from mangum import Mangum
+
+import feast
+from feast.feature_server import get_app
+
+# TODO: replace this with an actual feature repo config deserialized from the env variables
```
Comment (felixwang9817) [inline sdk/python/feast/infra/feature_servers/app.py:6]: I'm confused, why aren't we doing this now? Don't we want to build a Docker image that will actually construct the RepoConfig from environment variables?

Comment (tsotnet) [reply sdk/python/feast/infra/feature_servers/app.py:6]: Yes, that's a separate task though. I'll send out another PR for that (just trying to iterate quickly)

File: sdk/python/feast/infra/feature_servers/Dockerfile (hunk 1 above)
Comment (tsotnet) [reply sdk/python/feast/infra/feature_servers/Dockerfile:15]: iirc it was slowing things down. We can enable it after the issue with usage logging is fixed.

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**